lib/src/config_resolver.rs

```diff
@@ -64,6 +66,8 @@ pub struct ConfigResolutionContext<'a> {
 struct ScopeCondition {
     /// Paths to match the repository path prefix.
     pub repositories: Option<Vec<PathBuf>>,
+    /// Hostnames to match the hostname.
+    pub hostnames: Option<Vec<String>>,
```

Comment:
nit: Since `hostname` should be a static value, I would move it closer to `platforms`.